static/js/blueslip.js

```diff
@@ -96,20 +98,102 @@ exports.get_log = function blueslip_get_log() {
     return logger.get_log();
 };
 
-// Format error stacks using the ErrorStackParser
-// external library
-function getErrorStack(stack) {
+var stack_trace_gps = new StackTraceGPS();
+
+function beautify_stackframe(stackframe) {
+    return stack_trace_gps
+        .getMappedLocation(stackframe)
+        .then(function (info) {
```

Comment:
It might make sense to make this anonymous function extracted with a name, so that we can write some unit tests for its formatting logic.  It has a lot of tricky regexs/cases and seems like the kind of thing that's ideal for that sort of testing.